### PR TITLE
Jbwallace

### PR DESCRIPTION
**Pull request by jbwallace123**
An active work in progress. Checking to see if this is heading the right direction. 

Thank you!

### CODE REVIEW

File: workflow/pipeline/photometry.py [hunk 1]
```diff
@@ -98,45 +97,344 @@ def make(self, key):
...
         meta_info = {}
         try:
-            with open(meta_info_file.as_posix()) as f:
-                meta_info = tomli.loads(f.read())
+            with open(meta_info_file, "rb") as f:
+                meta_info = tomli.load(f)
         except FileNotFoundError:
             logger.info("meta info is missing")
-        light_source_name = meta_info.get("Fiber", {}).get("light_source", "")
+        light_source_name = meta_info.get("Experimental_Details").get("light_source")
```
Comment (sidhulyalkar) [inline workflow/pipeline/photometry.py:104]:
```suggestion
        light_source_name = meta_info.get("Experimental_Details", {}).get("light_source", "")
```

File: workflow/pipeline/photometry.py [hunk 2]
```diff
@@ -98,45 +97,344 @@ def make(self, key):
...
+        if len(list(photometry_dir.glob("processed*.mat"))) > 0:
+            data_format = "matlab_data"
+            matlab_data: dict = spio.loadmat(
+            next(photometry_dir.glob("processed*.mat")), simplify_cells=True)[list(matlab_data.keys())[3]]
+        elif len(list(photometry_dir.glob("*timeseries_2.mat"))) > 0:
+            data_format = "demux_matlab_data"
+            demux_matlab_data: list[dict] = spio.loadmat(
+                next(photometry_dir.glob("*timeseries_2.mat")), simplify_cells=True
+            )["timeSeries"]
+        else:
```
Comment (sidhulyalkar) [inline workflow/pipeline/photometry.py:119]: Here you could change to an elif, and verify that the TDT is present, before continuing.

File: workflow/pipeline/photometry.py [hunk 3]
```diff
@@ -98,45 +97,344 @@ def make(self, key):
...
+            data_format = "tdt_data"
+            tdt_data: tdt.StructType = tdt.read_block(photometry_dir)      
+        
+        ## Enter into different data format mode
+        if data_format == "matlab_data":
+            del matlab_data #write function here
+        elif data_format == "demux_matlab_data":
+            #demux_matlab_data
+            del demux_matlab_data
```
Comment (sidhulyalkar) [inline workflow/pipeline/photometry.py:128]: This looks like a typo, need to move the deletion of the data to the end of the if statement block, so the data can be accessed

**Comment by jbwallace123** [comment]
Made changes completing the last bits and adding new toml config

**Pull request closed without merging**